src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            {
+                Name = entityRestPath
+            };
+
+            // The OpenApiTag will categorize all paths created using the entity's name or overridden REST path value.
+            // The tag categorization will instruct OpenAPI document visualization tooling to display all generated paths together.
+            List<OpenApiTag> tags = new()
+            {
+                openApiTag
+            };
+
+            if (includePrimaryKeyPathComponent)
+            {
+                Tuple<string, List<OpenApiParameter>> pkComponents = CreatePrimaryKeyPathComponentAndParameters(entityName);
+                string pkPathComponents = pkComponents.Item1;
+                string fullPathComponent = entityBasePathComponent + pkPathComponents;
+
+                OpenApiOperation getOperation = new()
+                {
+                    Description = GETONE_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                };
+                getOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+
+                OpenApiOperation putOperation = new()
+                {
+                    Description = PUT_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
+                };
+                putOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+                putOperation.Responses.Add(HttpStatusCode.Created.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Created), responseObjectSchemaName: entityName));
+
+                OpenApiOperation patchOperation = new()
+                {
+                    Description = PATCH_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
+                };
+                patchOperation.Responses.Add(HttpStatusCode.OK.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.OK), responseObjectSchemaName: entityName));
+                patchOperation.Responses.Add(HttpStatusCode.Created.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Created), responseObjectSchemaName: entityName));
+
+                OpenApiOperation deleteOperation = new()
+                {
+                    Description = DELETE_DESCRIPTION,
+                    Tags = tags,
+                    Responses = new(_defaultOpenApiResponses),
+                    RequestBody = CreateOpenApiRequestBodyPayload(schemaReferenceId, requestBodyRequired)
```

Comment:
When is request body required for a delete operation?

Reply:
It is not allowed, removing.

Reply:
done